lib/galaxy/jobs/__init__.py

```diff
@@ -192,6 +192,9 @@ def __parse_job_conf_xml(self, tree):
                             else:
                                 self.handlers[tag] = [id]
 
+        # Must define at least one handler to have a default.
+        if not self.handlers:
+            raise ValueError("Job configuration file defines no valid handler elements.")
```

Comment:
Maybe `ConfigurationError` from `galaxy.exceptions` would be a better exception to raise?